src/textformatter.cpp

```diff
@@ -61,23 +61,36 @@ std::vector<std::string> wrap_line(
 	std::string curline = "";
 
 	for (auto word : words) {
-		size_t word_length = utils::strwidth_stfl(word);
-		size_t curline_length = utils::strwidth_stfl(curline);
-
-		// For words wider than the available width we have no other choice but
-		// to force splits at width limit
-		while (word_length > width) {
-			size_t space_left = width - curline_length;
-			curline.append(word.substr(0, space_left));
-			word.erase(0, space_left);
+		size_t word_width = utils::strwidth_stfl(word);
+		size_t curline_width = utils::strwidth_stfl(curline);
+
+		// for languages (e.g., CJK) don't use a space as a word boundary
+		while (word_width > width) {
+			size_t space_left = width - curline_width;
+			std::string part = utils::substr_with_width(word, space_left);
+			curline.append(part);
+			word.erase(0, part.length());
+			if (word.empty()) {
```

Comment:
Removing this check didn't break tests, so either the check is not necessary, or tests don't execute that code path at all. Can you please add such a test?

Reply:
Sure, I'll add a test (or remove this check). I thought that prefix/whitespace-only line
may be appeded to the result without this conditin, but I might be wrong.